File: day1_part2_2015.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").unwrap();
    let mut floor = 0;
    let mut position = 0;
    let mut basement_position = 0;

    for c in input.chars() {
        position += 1;
        if c == '(' {
            floor += 1;
        } else if c == ')' {
            floor -= 1;
        }

        if floor == -1 && basement_position == 0 {
            basement_position = position;
        }
    }

    println!("{}", floor);
    println!("{}", basement_position);
}

File: day1_part2_2015.java

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class solution {
    public static void main(String[] args) {
        try {
            BufferedReader reader = new BufferedReader(new FileReader("input.txt"));
            String input = reader.readLine().trim();
            reader.close();
            int floor = 0;
            int position = 0;
            for (int i = 0; i < input.length(); i++) {
                char c = input.charAt(i);
                if (c == '(') {
                    floor++;
                } else if (c == ')') {
                    floor--;
                }
                if (floor == -1) {
                    position = i + 1;
                    break;
                }
            }
            System.out.println(position);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
